### ISSUE TITLE
False positive "Octal file permissions must contain leading zero or be a string"

### ISSUE THREAD

**Issue by Nklya** [ansible-community/ansible-lint#566]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
ansible 2.8.3
  config file = None
  configured module search path = ['/Users/nklya/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/nklya/.pyenv/versions/3.7.2/lib/python3.7/site-packages/ansible
  executable location = /Users/nklya/.pyenv/versions/3.7.2/bin/ansible
  python version = 3.7.2 (default, Mar 17 2019, 13:02:41) [Clang 10.0.0 (clang-1000.10.44.4)]

ansible-lint --version
ansible-lint 4.1.1a0

I also checked v4.1.0 with the same result
```

- ansible installation method: pip / pyenv 3.7.2
- ansible-lint installation method: pip / pyenv 3.7.2

# Desired Behaviour

ansible-lint return no error for code from example below.

# Actual Behaviour (Bug report only)

I've got an error  `[202] Octal file permissions must contain leading zero or be a string` for this line of code in tasks.
I am sure that it's a false positive, because there is leading zero, and permissions are correct.

```yml
- name: False positive 1
  file:
    dest: "/opt/test"
    state: directory
    mode: 0701
```

Repository with test role - https://github.com/Nklya/test-octal

Log:

```sh
$ ansible-lint .
[202] Octal file permissions must contain leading zero or be a string
/Users/nklya/projects/edu/test-octal/tasks/main.yml:4
Task/Handler: False positive 1

[202] Octal file permissions must contain leading zero or be a string
/Users/nklya/projects/edu/test-octal/tasks/main.yml:10
Task/Handler: False positive 2
```

**Comment by ssbarnea** [comment]
[YAML specification](https://yaml.org/spec/1.2/spec.html) mentions `0o701` as valid and there is no false positive on it. Or you can assure it is a string as alternative.
